api/v1alpha3/types.go

```diff
@@ -327,18 +333,46 @@ type SubnetSpec struct {
 	Role SubnetRole `json:"role,omitempty"`
 
 	// ID defines a unique identifier to reference this resource.
+	// +optional
 	ID string `json:"id,omitempty"`
 
 	// Name defines a name for the subnet resource.
 	Name string `json:"name"`
 
 	// CidrBlock is the CIDR block to be used when the provider creates a managed Vnet.
+	// +optional
 	CidrBlock string `json:"cidrBlock,omitempty"`
 
 	// InternalLBIPAddress is the IP address that will be used as the internal LB private IP.
 	// For the control plane subnet only.
+	// +optional
 	InternalLBIPAddress string `json:"internalLBIPAddress,omitempty"`
 
 	// SecurityGroup defines the NSG (network security group) that should be attached to this subnet.
+	// +optional
 	SecurityGroup SecurityGroup `json:"securityGroup,omitempty"`
+
+	// RouteTable defines the route table that should be attached to this subnet.
+	// +optional
+	RouteTable RouteTable `json:"routeTable,omitempty"`
```

Comment:
Added `RouteTable` to make it easier to track and default the associated RouteTable in the spec. Also there's no reason why it shouldn't be configurable like security group is. For now it's just `Name` and `ID` but we can more fields later as needed. This is optional therefore not a breaking change. Will add conversion (noted as TODO in PR description).

Reply:
conversion added